src/loops/programs_with_repetition/easy/q4/MainTest.java

```diff
@@ -0,0 +1,86 @@
+package loops.programs_with_repetition.easy.q4;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import global.BaseTest;
+import global.exceptions.InvalidClauseException;
+import global.tools.TestOption;
+import global.variables.Clause;
+import global.variables.clauses.IntegerLiteral;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.PlaceHolder;
+import global.variables.clauses.StringLiteral;
+
+public class MainTest extends BaseTest {
```

Comment:
I think what you did here with the random input is a good idea; however because you did it randomly, there is a chance that a program which should not pass will if a negative or a number above 50 is not selected.

Reply:
Generally, we want to hardcode some base cases for things like boundaries and general logic, and then we can add a few random cases to catch hardcoding if necessary.